keras_nlp/models/bart/bart_seq_2_seq_lm_preprocessor.py

```diff
@@ -185,3 +187,104 @@ def call(self, x, y=None, sample_weight=None):
         y = decoder_token_ids[..., 1:]
         sample_weight = decoder_padding_mask[..., 1:]
         return pack_x_y_sample_weight(x, y, sample_weight)
+
+    def generate_preprocess(
+        self,
+        x,
+        sequence_length=None,
+    ):
+        """Convert encoder and decoder input strings to integer token inputs for generation.
+
+        Similar to calling the layer for training, this method takes in a dict
+        containing `"encoder_text"` and `"decoder_text"`, with strings or tensor
+        strings for values, tokenizes and packs the input, and computes a padding
+        mask masking all inputs not filled in with a padded value.
+
+        For the encoder inputs, the above step is exactly the same as for
```

Comment:
Unlike calling the the layer for training, this method does not compute
labels and will never append a `tokenizer.end_token_id` to the end of
the decoder sequence (as generation is expected to continue at the end
of the inputted decoder prompt).